server/game/cards/locations/04/harrenhalffh.js

```diff
@@ -1,38 +1,36 @@
 const DrawCard = require('../../../drawcard.js');
 
 class Harrenhal extends DrawCard {
-    setupCardAbilities() {
+    setupCardAbilities(ability) {
         this.action({
             title: 'Kneel Harrenhal to put a character in play',
-            method: 'kneel',
-            phase: 'challenge'
+            cost: ability.costs.kneelSelf(),
+            phase: 'challenge',
+            handler: context => {
```

Comment:
I'm not super thrilled about how much this has increased the indentation vs separate methods